src/charm.py

```diff
@@ -175,6 +188,30 @@ def _on_orchestrator_available(self, event: OrchestratorAvailableEvent):
             return
         self.unit.status = ActiveStatus()
 
+    def _on_lte_core_relation_joined(self, event: RelationJoinedEvent):
+        """Triggered when lte-core relation is joined.
+
+        AGW will provide the IP address of the MME (eth1) interface if that is available.
+
+        Args:
+            event: Juju event (RelationJoinedEvent)
+
+        Returns:
+            None
+        """
+        if not self.unit.is_leader():
+            return
+        try:
+            ip = netifaces.ifaddresses("eth1")[netifaces.AF_INET][0]["addr"]
+            self._lte_core_provides.set_lte_core_information(ip)
+            self.unit.status = ActiveStatus()
+        except (ValueError, AddressValueError) as e:
+            logger.warning("Failed to fetch IP address of eth1 interface")
+            logger.error(str(e))
```

Comment:
```suggestion
            logger.error(f"Failed to fetch IP address of eth1 interface: {str(e)}")
```
How about something like that?